.github/workflows/ci.yml

```
@@ -0,0 +1,101 @@
+
+        steps:
+            -   name: Checkout source
+                uses: actions/checkout@v2
+
+            -   name: Get code coverage driver
+                uses: actions/github-script@v3.1
+                id: code-coverage-driver
+                with:
+                    script: 'return "${{ matrix.calculate-code-coverage }}" == "yes" ? "pcov" : "none"'
+                    result-encoding: string
+
+            -   name: Get symfony flex version
+                uses: actions/github-script@v3.1
+                id: flex-version
+                with:
+                    script: 'return "${{ matrix.php }}" == "7.0" ? "~1.6.0" : "^1.11"'
+                    result-encoding: string
+
+            -   name: PHP setup
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: ${{ matrix.php }}
+                    coverage: ${{ steps.code-coverage-driver.outputs.result }}
+
+            -   name: Downgrade composer for olf Symfony Flex
+                if: steps.flex-version.outputs.result == '~1.6.0'
+                run: composer self-update --1
+
+            -   name: Install Symfony Flex
+                if: startsWith(matrix.php, '5') == false
+                run: composer global require -o --no-interaction symfony/flex:${{ steps.flex-version.outputs.result }}
+
+            -   name: Require Symfony version
+                if: matrix.symfony
+                run: composer config extra.symfony.require ${{ matrix.symfony }}
+
+            -   name: Lower minimum stability to test with Symfony 6
+                if: matrix.symfony == '^6.0'
+                run: composer config minimum-stability beta
+
+            -   name: Update composer dependencies
+                run: composer update -o --no-interaction --no-progress ${{ matrix.composer-flags }}
+
+            -   name: Run test suite
+                run: ./vendor/bin/phpunit -v
+
+            -   name: Upload coverage results to Coveralls
+                if: matrix.calculate-code-coverage == 'yes'
+                env:
+                    COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                run: bin/php-coveralls -v --exclude-no-stmt
+    build-phar:
+        needs: tests
+        runs-on: ubuntu-latest
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: 7.0
+                    coverage: none
+
+            -   name: Build PHAR
+                run: |
+                    curl -LSs https://box-project.github.io/box2/installer.php | php
+                    composer config platform.php 2> /dev/null || composer config platform.php 5.5.0
+                    composer update --no-dev --no-interaction --optimize-autoloader --no-interaction --no-progress
+                    php -d phar.readonly=false box.phar build
```

```suggestion
                    php -d phar.readonly=false box.phar build
                    ./build/artifacts/php-coveralls.phar --version
```


very base and naive Smoke test that built phar file is runnable and not crashing